contracts/Chorus.sol

```diff
@@ -66,206 +81,155 @@ contract Chorus is Inflation, OracleGetter, ERC20 {
         OracleGetter(_tellorAddress)
         ERC20(_tokenName, _tokenSymbol)
         within100e18Range(_inflRatePerYear)
-        within1e18Range(_collateralPriceGranularity)
     {
-        // TODO Check if token ID is supported by the oracle and returns the price in USD.
-        // For now assume that the contract creator knows what he is doing.
-        // The collateral ID needs to return the value in USD.
         collateralID = _collateralID;
         collateralToken = ERC20(_collateralToken);
         collateralPriceGranularity = _collateralPriceGranularity;
-
+        require(_collateralPriceGranularity > 0 && _collateralPriceGranularity <= 1e18, "value not within allowed limits");
         require(_inflBeneficiary != address(0), "benificiary address not set");
         inflBeneficiary = _inflBeneficiary;
         inflRatePerSec = yearlyRateToPerSec(_inflRatePerYear);
     }
 
-    modifier onlyAdmin {
-        require(msg.sender == admin, "not an admin");
-        _;
-    }
-
-    modifier within100e18Range(uint256 _value) {
-        require(_value > 0 && _value < 100e18, "value not within allowed limits");
-        _;
+    /**
+     * @dev Returns the current token price in USD reduced by the current inflation.
+     * @return uint256 token price
+     */
+    function collateralBalance() external view returns (uint256) {
+        return collateralToken.balanceOf(address(this));
     }
 
-    modifier within1e18Range(uint256 _value) {
-        require(_value > 0 && _value <= 1e18, "value not within allowed limits");
-        _;
+    /**
+     * @dev Checks the Tellor oracle and gets the collateral price
+     * @return uint256 collateral price in USD upscaled to e18 precision.
+     */
+    // slither-disable-next-line timestamp
+    function collateralPrice() public view returns (uint256) {
+        (bool _didGet, uint256 _collateralPrice, ) =
+            _getDataBefore(collateralID, block.timestamp - collateralPriceAge);
+        require(_didGet, "getting oracle price");
+        return mul(_collateralPrice, div(1e18, collateralPriceGranularity));
     }
 
-    function setAdmin(address _newAdmin) external onlyAdmin {
-        require(_newAdmin != address(0), "cannot send to the zero address");
-        admin = _newAdmin;
-        emit NewAdmin(_newAdmin);
+    /**
+     * @dev A view funciton to look at the collateralization of the system
+     * @return uint256 collateral token balance of this address / totalSupply
+     */
+    function collateralRatio() public view returns (uint256) {
+        uint256 _collateralBalance = collateralToken.balanceOf(address(this));
+        // slither-disable-next-line dangerous-strict-equalities
```

Comment:
I think this is the way to disable the check https://github.com/crytic/slither/wiki/Detector-Documentation#dangerous-strict-equalities
```suggestion
        // slither-disable-next-line incorrect-equality
```